dns_entry_checker/dns_entry_checker.py

```diff
@@ -0,0 +1,223 @@
+        with open(backward_missing_filepath, "a") as backward_missing_file:
+            backward_missing_file.write(f"{ips_dns_pair[1]}\n")
+    elif not backward_ip[0][0] == ips_dns_pair[0]:
+        with open(backward_mismatch_filepath, "a") as backward_mismatch_file:
+            backward_mismatch_file.write(f"{ips_dns_pair[1]}\n")
+
+
+def check_missing_ips(key, gap_missing_filepath):
+    """
+    Function to check which IPs are missing from 2-254 and output them to a file
+        :param key: List containing (List)
+        :param gap_missing_filepath: The filepath of the gap missing output file (String)
+    """
+    res = set([eval(j) for j in key[1]])
+    missing_values = [k for k in (range(2, 255)) if k not in res]
+    for ip_last_byte in missing_values:
+        with open(gap_missing_filepath, "a") as gap_missing_file:
+            gap_missing_file.write(f"172.16.{key[0]}.{ip_last_byte}\n")
+
+
+def check_output_files(
+    forward_mismatch_filepath,
+    backward_mismatch_filepath,
+    backward_missing_filepath,
+    gap_missing_filepath,
+):
+    """
+    Function to check if output files already exist
+        :param forward_mismatch_filepath: The filepath of the forward mismatch output file (String)
+        :param backward_mismatch_filepath: The filepath of the backward mismatch output file (String)
+        :param backward_missing_filepath: The filepath of the backward missing output file (String)
+        :param gap_missing_filepath: The filepath of the gap missing output file (String)
+    """
+    if os.path.exists(forward_mismatch_filepath):
+        raise RuntimeError(f"{forward_mismatch_filepath} already exists")
+    if os.path.exists(backward_mismatch_filepath):
+        raise RuntimeError(f"{backward_mismatch_filepath} already exists")
+    if os.path.exists(backward_missing_filepath):
+        raise RuntimeError(f"{backward_missing_filepath} already exists")
+    if os.path.exists(gap_missing_filepath):
+        raise RuntimeError(f"{gap_missing_filepath} already exists")
+
+
+def dns_entry_checker():
+    # Get arguments passed to the script
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument(
+        "-u", "--user", metavar="USER", help="FedID of the user", required=True
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        metavar="PASSWORD",
+        help="Password of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-i",
+        "--ip",
+        metavar="IP",
+        help="IP of the machine to SSH to",
+        required=True,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="OUTPUT",
+        help="Directory to create the output files in",
+    )
+
+    # Define the variables with the script arguments
+    args = parser.parse_args()
+    user = args.user
+    password = args.password
+    ip = args.ip
+    output = args.output
```

Comment:
you can move this into another function - and return a dictionary of argument names : argument values 
```
def parse_args(inp_args):
    """
    Function to parse commandline args
    :param inp_args: a set of commandline args to parse (dict)
    :returns: A dictionary of parsed args 
    """
    ...code here..
    args = parser.parse_args(inp_args)
    return args
```
you can call the function like so
```
args = parse_args(sys.argv[1:])